packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -168,9 +183,27 @@ - (instancetype)initWithCameraName:(NSString *)cameraName
   _motionManager = [[CMMotionManager alloc] init];
   [_motionManager startAccelerometerUpdates];
 
-  if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
-    return nil;
+  NSError *outError;
+  if ([_captureDevice lockForConfiguration:&outError]) {
+    [_videoCaptureSession beginConfiguration];
+
+    if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
+      [_videoCaptureSession commitConfiguration];
+      [_captureDevice unlockForConfiguration];
+      return nil;
+    }
+
+    if (_fps) {
+      _captureDevice.activeVideoMinFrameDuration = CMTimeMake(1, [_fps intValue]);
+      _captureDevice.activeVideoMaxFrameDuration = CMTimeMake(1, [_fps intValue]);
+    }
+
+    [_videoCaptureSession commitConfiguration];
+    [_captureDevice unlockForConfiguration];
+  } else {
+    NSLog(@"error locking device for frame rate change (%@)", outError);
```

Comment:
do we want to surface the `outError` to dart side?

Reply:
the outError surfaced to dart side

Reply:
surfacing if not `nil`. actually test device implementation returns NO with `outError == nil` on lock at this place.